### PR TITLE
Add projectName Formatter

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
You might name a project `name = cool-project`. However if you then use this in your `build.sbt` you will end up not compiling

```scala
lazy val $name$ = project.in(....)
output
lazy val cool-project = project.in(...) // Invalid
```

I propose adding projectName so we can generate from the formatter

```scala
lazy val $name;format="projectName"$ = project.in(...)
would expand to
lazy val `cool-project` = project.in(...)
```

I am uncertain if this will meet my end goal as I lack familiarity with the codebase. Hopefully I am atleast on the right track.

### CODE REVIEW

**Comment by ChristopherDavenport** [comment]
On second thought, its more of a dashed variable encapsulation than merely project name. But I will wait on thoughts. :smile:

File: library/src/main/scala/g8.scala [hunk 1]
```diff
@@ -207,7 +207,8 @@ object G8 {
   def snakeCase(s: String)    = s.replaceAll("""[\s\.\-]+""", "_")
   def packageDir(s: String)   = s.replace(".", System.getProperty("file.separator"))
   def addRandomId(s: String)  = s + "-" + new java.math.BigInteger(256, new java.security.SecureRandom).toString(32)
-
+  def projectName(s: String)  = if (s.contains("-")) "`" + s + "`" else s
```
Comment (eed3si9n) [inline library/src/main/scala/g8.scala]: Not sure if "projectName" is the best name for the formatter.
Maybe `scalaIdentifier`?

**Comment by ChristopherDavenport** [comment]
Updated per review comment.

**Comment by eed3si9n** [comment]
@ChristopherDavenport All I see is "Merge branch 'master..."

**Comment by ChristopherDavenport** [comment]
That's weird. I thought I pushed first, then saw a complaint about the branch so clicked the gh fix button. Apparently I forgot the pushing step. My apologies will fix.

**Comment by ChristopherDavenport** [comment]
Alright, sorry about the bad update. I accidentally made a new branch that I updated rather than the one in this PR. Looks ok now, and should be updated as requested.

**Pull request merged**